src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,305 @@
+            $permissions,
+            $permissionIds,
+            $employeeProfileId,
+            $canEmployeeEditPermissions
+        );
+    }
+
+    /**
+     * @return int[] IDs of non configurable tabs
+     */
+    private function getNonConfigurableTabs()
+    {
+        return [
+            Tab::getIdFromClassName('AdminLogin'),
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    private function getProfilesForPermissionsConfiguration()
+    {
+        $legacyProfiles = Profile::getProfiles(Context::getContext()->language->id);
+        $profiles = [];
+
+        foreach ($legacyProfiles as $profile) {
+            $isAdministrator = (int) $profile['id_profile'] === _PS_ADMIN_PROFILE_;
+
+            $profiles[] = [
+                'id' => $profile['id_profile'],
+                'name' => $profile['name'],
+                'is_administrator' => $isAdministrator,
+            ];
+        }
+
+        return $profiles;
+    }
+
+    /**
+     * @return array
+     */
+    private function getTabsForPermissionsConfiguration()
+    {
+        $nonConfigurableTabs = $this->getNonConfigurableTabs();
+        $legacyTabs = Tab::getTabs(Context::getContext()->language->id);
+        $tabs = [];
+
+        foreach ($legacyTabs as $tab) {
+            // Don't allow permissions for unnamed tabs (ie. AdminLogin)
+            if (empty($tab['name'])) {
+                continue;
+            }
+
+            if (in_array($tab['id_tab'], $nonConfigurableTabs)) {
+                continue;
+            }
+
+            $tabs[] = [
+                'id' => $tab['id_tab'],
+                'id_parent' => $tab['id_parent'],
+                'name' => $tab['name'],
+            ];
+        }
+
+        return $this->buildTabsTree($tabs);
+    }
+
+    /**
+     * @param array $tabs
+     * @param int $parentId
+     *
+     * @return array
+     */
+    private function buildTabsTree(array &$tabs, $parentId = 0)
+    {
+        $children = [];
+
+        foreach ($tabs as &$tab) {
+            $id = $tab['id'];
+
+            if ((int) $tab['id_parent'] === (int) $parentId) {
+                $children[$id] = $tab;
+                $children[$id]['children'] = $this->buildTabsTree($tabs, $id);
+            }
+        }
+
+        return $children;
+    }
+
+    /**
+     * @param array $profiles
+     *
+     * @return array
+     */
+    private function getTabPermissionsForProfiles(array $profiles)
+    {
+        $permissions = [];
+
+        foreach ($profiles as $profile) {
+            $permissions[$profile['id']] = Profile::getProfileAccesses($profile['id']);
+        }
+
+        return $permissions;
+    }
+
+    /**
+     * @param int $employeeProfileId
+     * @param bool $hasEmployeeEditPermission
+     * @param array $profileTabPermissions
+     * @param array $profiles
+     * @param array $tabs
+     * @param array $permissions
+     *
+     * @return array
+     */
+    private function getBulkConfigurationForProfiles(
```

Comment:
this methods contains some logic (:hankey:) extracted from legacy template, i _may_ improve it later on.

Reply:
👍